scripts/ebeam_scan_to_multiprobe.py

```diff
@@ -0,0 +1,94 @@
+import numpy
+import sys
+from builtins import input
+
+from odemis.acq.align.spot import FindGridSpots
+from odemis.driver import ueye
+
+
+def get_deflection_angle(coordinates):
+    """
+    Fit a line through a set of coordinates and calculate the angle between that line and the x-axis, this is the
+    deflection angle.
+
+    coordinates (ndarray of shape nx2): (x, y) coordinates of center positions of the grid of spots.
+    return:
+        phi (float): deflection angle in radians.
+    """
+    # Construct a system of linear equations to fit the line a*x + b*y + c = 0
+    n = coordinates.shape[0]
+    if n < 2:
+        raise ValueError("n should be higher than or equal to 2, cannot find deflection angle for n is {}".format(n))
+    elif n == 2:
+        x = coordinates[1, :] - coordinates[0, :]
+        # Determine the angle of this line to the x-axis, -pi/2 < phi < pi/2
+        phi = numpy.arctan2(x[0], x[1])
+    else:
+        A = numpy.hstack((coordinates, numpy.ones((n, 1))))
+
+        # Solve the equation A*x = 0; i.e. find the null space of A. The solution is
+        # the eigenvector corresponding to the smallest singular value.
+        U, s, V = numpy.linalg.svd(A, full_matrices=False)
+        x = numpy.transpose(V[-1])
+
+        # Determine the angle of this line to the x-axis, -pi/2 < phi < pi/2
+        phi = numpy.arctan2(-x[0], x[1])
+    phi = (phi + math.pi / 2) % math.pi - math.pi / 2
+    return phi
+
+
+def main(args):
+    """
+    Handles the command line arguments
+    args is the list of arguments passed
+    return (int): value to return to the OS as program exit code
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--debug', action="store_true", default=False)
+
+    options = parser.parse_args(args[1:])
+    if options.debug:
+        logging.getLogger().setLevel(logging.DEBUG)
+    else:
+        logging.getLogger().setLevel(logging.WARNING)
+
+    ccd = ueye.Camera("camera", "ccd", device=None)
+    image = ccd.data.get(asap=False)
+    n_spots = (8, 8)
+    spot_coordinates, translation, scaling, rotation = FindGridSpots(image, n_spots)
+    coordinates = [translation]
+    input('Press enter after applying x-shift to electron beam: \n')
+    while True:
+        image = ccd.data.get(asap=False)
+        spot_coordinates, translation, scaling, rotation = FindGridSpots(image, n_spots)
+        coordinates.append(translation)
+        inp = input('Applied x-shift to electron beam again? y/n \n')
+        while inp.lower() not in ['y', 'n']:
+            inp = input('{} not an option please choose from y/n \n'.format(inp))
+        if inp.lower() == "n":
+            break
+
+    deflection_angle = get_deflection_angle(numpy.array(coordinates))
+    sem_rotation = deflection_angle - rotation
+    print("Apply {:.3f}° to the scan rotation using SEM PC".format(math.degrees(sem_rotation)))
+    return 0
+
+
+if __name__ == '__main__':
+    ret = main(sys.argv)
+    exit(ret)
```

Comment:
@pieleric when I run the script through the terminal on Ubuntu 18.04 it does not exit, a keyboard interrupt also doesn't work. Do you know if I should use something else to exit?